### PR TITLE
feat: make IAM role an optional parameter for EcsIsoServiceAutoscalerProps

### LINKED ISSUE #41: feat: make role an optional parameter for EcsIsoServiceAutoscalerProps
### Description

Currently you must provide an IAM role to assign to the lambda function that handles autoscaling in the `EcsIsoServiceAutoscaler` construct.

I'd like to make the `role` property optional, allowing the `Function` construct to create it's own IAM role, and for the `Function` to be exposed as a property of `EcsIsoServiceAutoscaler`

### Use Case

Within the enterprise, I'm able to create my own IAM roles, so want to allow the `Function` construct to create the role on my behalf

### Proposed Solution

Turn the following property of `EcsIsoServiceAutoscalerProps` from 

```ts
readonly role: IRole;
```

to 

```ts
readonly role?: IRole;
```

Ensure that the `ecsScalingManager` function is a public readonly property of the construct so it's accessible from outside the construct

### Other information

_No response_

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by maafk**
Fixes #41 

Makes it optional to provide a role in `EcsIsoServiceAutoscalerProps`

If a role is _not_ provided, we let the Lambda function create the role, and ensure that role has least privilege permissions

### CODE REVIEW

File: src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts [hunk 1]
```diff
@@ -9,16 +9,46 @@ import { Alarm } from 'aws-cdk-lib/aws-cloudwatch';
...
 import { Construct } from 'constructs';
 
 export interface EcsIsoServiceAutoscalerProps {
   /**
-   * The IAM role that allows the created lambda to adjust the desired count on the ECS Service .
-   * TODO: Make optional and add function for auto generation
+   * Optional IAM role to attach to the created lambda to adjust the desired count on the ECS Service
+   *
+   * Ensure this role has appropriate privileges. Example IAM policy statements:
+   * And more
```
Comment (joel-aws) [inline src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts:21]: This seems out of place.

File: src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts [hunk 2]
```diff
@@ -9,16 +9,46 @@ import { Alarm } from 'aws-cdk-lib/aws-cloudwatch';
...
+   *          }
+   *        },
+   *        "Effect": "Allow",
+   *        "Resource": "arn:${Partition}:ecs:${Region}:${Account}:service/${ClusterName}/${ServiceName}"
+   *      }
+   *    ],
+   *    "Version": "2012-10-17"
+   *  }
+   *}
+   * ```
```
Comment (joel-aws) [inline src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts:48]: Would a CDK-native version of a [Role](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.Role.html) with the right [PolicyDocument](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.PolicyDocument.html) more sense?

File: src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts [hunk 3]
```diff
@@ -127,7 +159,31 @@ export class EcsIsoServiceAutoscaler extends Construct {
       description: `Kicks off Lambda to adjust ECS scaling for service: ${props.ecsService.serviceName}`,
       enabled: true,
       schedule: Schedule.rate(Duration.minutes(1)),
-      targets: [new LambdaFunction(ecsScalingManager)],
+      targets: [new LambdaFunction(this.ecsScalingManagerFunction)],
     });
+
+    if (!props.role) {
```
Comment (joel-aws) [inline src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts:165]: Would also be nice if the role was exposed as a property.

E.g.,

```
this.role = props.role || new iam.Role(this, 'ServiceRole', {
  assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
  managedPolicies,
});
```

**Review by joel-aws — CHANGES_REQUESTED** [review]

File: src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts (hunk 2 above)
Comment (maafk) [reply src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts:48]: The original implementation had a role passed into it to attach to the Lambda function that handled the autoscaling.  The idea being that the enterprise didn't allow developers to create IAM roles, so they'd have to pass a provided role in.

This explanation of what IAM policy is needed for the Lambda function to do what it needs is more for a developer to provide to a centralized team in the enterprise who creates these roles for them

File: src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts (hunk 1 above)
Comment (maafk) [reply src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts:21]: The idea is that if a developer needs for a separate team in the enterprise to create a role on their behalf to make sure it has sufficient privileges to do it's autoscaling job

File: src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts (hunk 3 above)
Comment (maafk) [reply src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts:165]: Part of the change in this PR includes [exposing](https://github.com/cdklabs/cdk-enterprise-iac/pull/43/files#diff-6a3b40df0eff41157e397f3ca8f7d63a4450e11870a8928941b997afeb0bf6eeR131) the lambda function itself, so the role should already be available in `this.ecsScalingManagerFunction.role`

**Comment by joel-aws** [comment]
Looks good!

**Review by joel-aws — APPROVED** [review]

**Pull request merged**